pkg/shp/cmd/buildrun/cancel.go

```diff
@@ -0,0 +1,74 @@
+package buildrun
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/cli/pkg/shp/cmd/runner"
+	"github.com/shipwright-io/cli/pkg/shp/params"
+	"github.com/shipwright-io/cli/pkg/shp/resource"
+)
+
+// CancelCommand contains data input from user for delete sub-command
+type CancelCommand struct {
+	cmd *cobra.Command
+
+	name string
+}
+
+func cancelCmd() runner.SubCommand {
+	return &CancelCommand{
+		cmd: &cobra.Command{
+			Use:   "cancel <name>",
+			Short: "Cancel BuildRun",
+			Args:  cobra.ExactArgs(1),
+		},
+	}
+}
+
+// Cmd returns cobra command object
+func (c *CancelCommand) Cmd() *cobra.Command {
+	return c.cmd
+}
+
+// Complete fills in data provided by user
+func (c *CancelCommand) Complete(params *params.Params, args []string) error {
+	c.name = args[0]
+
+	return nil
+}
+
+// Validate validates data input by user
+func (c *CancelCommand) Validate() error {
+	return nil
```

Comment:
Can a BuildRun name be any length and contain any characters? Or does it need to adhere to some kind of label rules or regexp?

Reply:
that's the type of validation to be done on the generated code side in shipwright-io/build ... leaving this as is

as a point of comparison, while there are analogous Run and Complete semantics over in openshift/oc, there is not "Validate" one.

Looking at the other verbs here, I see one instance where we check for the name being empty, but it seems like the arg check should over that

Feels like a separate cli issue of (what should we do with Validate / should we keep Validate)

Not pulling that into this PR